### PR TITLE
supportsHibernation property added

### PR DESCRIPTION
**Pull request by meet1993shah**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json [hunk 1]
```diff
@@ -2369,6 +2380,10 @@
         "purchasePlan": {
           "$ref": "#/definitions/PurchasePlan",
           "description": "Purchase plan information to be added on the OS disk"
+        },
+        "supportsHibernation": {
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json:2384]: needs to be present in SnapshotUpdateProperties too

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json [hunk 2]
```diff
@@ -3103,6 +3118,10 @@
           "$ref": "#/definitions/Encryption",
           "description": "Encryption property can be used to encrypt data at rest with customer managed keys or platform managed keys.",
           "readOnly": true
+        },
+        "supportsHibernation": {
+          "type": "boolean",
+          "description": "Indicates the OS on a particular disk supports hibernation."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json:3124]: for the property in snapshot please update the description.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/examples/UpdateAManagedDiskToAddSupportsHibernation.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        "name": "myDisk",
+        "location": "westus",
+        "sku": {
+          "name": "Standard_LRS",
+          "tier": "Standard"
+        },
+        "properties": {
+          "osType": "Windows",
+          "hyperVGeneration": "V1",
+          "supportsHibernation": true,
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/examples/UpdateAManagedDiskToAddSupportsHibernation.json:25]: FromImage wouldnt we infer the property?
shouldnt this be empty disk or Import?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json [hunk 4]
```diff
@@ -1986,6 +1989,10 @@
         "burstingEnabled": {
           "type": "boolean",
           "description": "Set to true to enable bursting beyond the provisioned performance target of the disk. Bursting is disabled by default. Does not apply to Ultra disks."
+        },
+        "supportsHibernation": {
+          "type": "boolean",
+          "description": "Indicates the OS on a particular disk supports hibernation."
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json:1995]: nit. Drop 'particular'

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/examples/UpdateAManagedDiskToAddSupportsHibernation.json (hunk 3 above)
Comment (vikramd-ms) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/examples/UpdateAManagedDiskToAddSupportsHibernation.json:25]: Import is more realistic

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json [hunk 5]
```diff
@@ -171,6 +171,9 @@
           },
           "Create or update a bursting enabled managed disk.": {
             "$ref": "./examples/CreateOrUpdateABurstingEnabledManagedDisk.json"
```
Comment (vikramd-ms) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json:173]: What about the new disk state - Frozen?

Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json:173]: oh, I totally forgot about the frozen tests, I will add those now.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/examples/UpdateAManagedDiskToAddSupportsHibernation.json (hunk 3 above)
Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/examples/UpdateAManagedDiskToAddSupportsHibernation.json:25]: Will remove this and add it for Import test

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json (hunk 5 above)
Comment (meet1993shah) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json:173]: as discussed, will add the examples related to disk state Frozen in next swagger version

**Pull request merged**